### PR TITLE
Added mean_absolute_error to tensorflow.metrics frontend fixed

### PR DESCRIPTION
**Pull request by AhmedMousa-ag**

### CODE REVIEW

**Comment by Daniel4078** [comment]
Hello and please fix the the two files conflicing with the main branch. After that I can see the status of the automated tests and give you my review. Thank you.

**Review by Daniel4078 — CHANGES_REQUESTED** [review]
Please first fix all these formatting errors given below, thank you!
ivy/functional/frontends/tensorflow/metrics.py:24:55: E126 continuation line over-indented for hanging indent
ivy/functional/frontends/tensorflow/metrics.py:24:74: E231 missing whitespace after ','
ivy/functional/frontends/tensorflow/metrics.py:24:89: E501 line too long (108 > 88 characters)
ivy/functional/frontends/tensorflow/metrics.py:25:55: E123 closing bracket does not match indentation of opening bracket's line
ivy/functional/frontends/tensorflow/metrics.py:27:1: E302 expected 2 blank lines, found 1
ivy/functional/frontends/tensorflow/metrics.py:28:35: E226 missing whitespace around arithmetic operator
ivy/functional/frontends/tensorflow/metrics.py:30:1: E305 expected 2 blank lines after class or function definition, found 1
ivy/functional/frontends/tensorflow/metrics.py:31:43: E126 continuation line over-indented for hanging indent
ivy/functional/frontends/tensorflow/metrics.py:31:62: E231 missing whitespace after ','
ivy/functional/frontends/tensorflow/metrics.py:31:89: E501 line too long (96 > 88 characters)
ivy/functional/frontends/tensorflow/metrics.py:32:43: E123 closing bracket does not match indentation of opening bracket's line
ivy_tests/test_ivy/test_frontends/test_tensorflow/test_metrics.py:177:1: E302 expected 2 blank lines, found 1
ivy_tests/test_ivy/test_frontends/test_tensorflow/test_metrics.py:191:1: E304 blank lines found after function decorator
ivy_tests/test_ivy/test_frontends/test_tensorflow/test_metrics.py:192:5: E125 continuation line with same indent as next logical line
ivy_tests/test_ivy/test_frontends/test_tensorflow/test_metrics.py:192:11: E231 missing whitespace after ','
ivy_tests/test_ivy/test_frontends/test_tensorflow/test_metrics.py:192:24: E231 missing whitespace after ','
ivy_tests/test_ivy/test_frontends/test_tensorflow/test_metrics.py:192:36: E231 missing whitespace after ','
ivy_tests/test_ivy/test_frontends/test_tensorflow/test_metrics.py:192:56: E231 missing whitespace after ','
ivy_tests/test_ivy/test_frontends/test_tensorflow/test_metrics.py:192:69: E231 missing whitespace after ','
ivy_tests/test_ivy/test_frontends/test_tensorflow/test_metrics.py:208:6: W292 no newline at end of file

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_metrics.py [hunk 1]
```diff
@@ -172,3 +172,36 @@ def test_sparse_categorical_crossentropy(
...
         from_logits=from_logits,
     )
+
+
+# mean_absolute_error
+@handle_cmd_line_args
+@given(
+    y_true=st.lists(st.integers(min_value=0, max_value=4), min_size=1, max_size=1),
+    dtype_y_pred=helpers.dtype_and_values(
+        available_dtypes=ivy_tf.valid_float_dtypes,
```
Comment (Daniel4078) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_metrics.py:182]: please use ivy.current_backend().valid_dtypes inside our helper functions

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_metrics.py [hunk 2]
```diff
@@ -172,3 +172,36 @@ def test_sparse_categorical_crossentropy(
...
+)
+def test_mean_absolute_error(
+        y_true, dtype_y_pred, as_variable, num_positional_args, native_array, fw):
+    y_true = ivy.array(y_true, dtype=ivy.int32)
+    dtype, y_pred = dtype_y_pred
+
+    helpers.test_frontend_function(
+        input_dtypes=[ivy.int32, dtype],
+        as_variable_flags=as_variable,
+        with_out=False,
```
Comment (Daniel4078) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_metrics.py:199]: with_out flag should be passed in as an argument of test_mean_absolute_error as it is handle by the handle_cmd_line_args as well.

**Review by Daniel4078 — CHANGES_REQUESTED** [review]
Thank you for fixing those formatting errors, now please make a few change to your test as described by my comments below.

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_metrics.py [hunk 3]
```diff
@@ -172,3 +172,36 @@ def test_sparse_categorical_crossentropy(
...
+    ),
+)
+def test_mean_absolute_error(
+        y_true, dtype_y_pred, as_variable, num_positional_args, native_array, fw):
+    y_true = ivy.array(y_true, dtype=ivy.int32)
+    dtype, y_pred = dtype_y_pred
+
+    helpers.test_frontend_function(
+        input_dtypes=[ivy.int32, dtype],
+        as_variable=as_variable,
```
Comment (Daniel4078) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_metrics.py:198]: this keyword_only argument's name is as_variable_flags not as_variable

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_metrics.py (hunk 2 above)
Comment (Daniel4078) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_metrics.py:199]: you should have a with_out argument for test_mean_absolute_error and let with_out=with_out here.

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_metrics.py [hunk 4]
```diff
@@ -172,3 +172,36 @@ def test_sparse_categorical_crossentropy(
...
         from_logits=from_logits,
     )
+
+
+# mean_absolute_error
+@handle_cmd_line_args
+@given(
+    y_true=st.lists(st.floats(min_value=0., max_value=4.), min_size=1, max_size=1),
+    dtype_y_pred=helpers.dtype_and_values(
+        available_dtypes=ivy_tf.valid_float_dtypes,
```
Comment (Daniel4078) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_metrics.py:182]: why did you use ivy_tf here instead of ivy.current_backend()?

Comment (AhmedMousa-ag) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_metrics.py:182]: I was testing the code, and in Pycharm it passes all the tests, I'm not sure what's wrong here

**Comment by Daniel4078** [comment]
Currently, there is only one error to be fixed about your code:
AssertionError: the return with a TensorFlow backend produced data type of float32, while the return with a tensorflow backend returned a data type of float64.
Thank you.

**Review by Daniel4078 — CHANGES_REQUESTED** [review]
I checked the document of mean_absolute_error, the shape of y_true and y_pred that you generate for the test should be the same.

**Comment by AhmedMousa-ag** [comment]
> the shape of y_true and y_pred that you generate for the test should be the same.

I modified the test function to make share it generate values of the same type and shape for y_true and y_pred, had a problem with types int8 and float64 in NumPy and torch even though I used ivy.current_backend().valid_dtypes "I checked the backend_handler.py, it doesn't have valid_dtypes attribute"
So I had to use the unsupported_types function

**Review by Daniel4078 — CHANGES_REQUESTED** [review]
Well done, now it seems that your test is passing. Just one small change, please add a with_out argument for the test_mean_absolute_error function and pass tha argument to the with_out argument of the test_frontend_function as that flag show be set by the decorator handle_cmd_lines_args automatically instead of bing fixed to False. Thank you!

**Review by Daniel4078 — APPROVED** [review]
I think this pull request is good to go.

**Pull request merged**